components/properties/src/props.rs

```diff
@@ -38,6 +40,38 @@ pub enum EnumeratedProperty {
     InvalidCode = -1, // TODO(#1160) - taken from ICU4C UProperty::UCHAR_INVALID_CODE
 }
 
+/// Enumerated property Bidi_Class
+///
+///
+///
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Ord, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum BidiClass {
```

Comment:
suggestion: make `BidiClass` a newtype, not a Rust enum

Rust enums are defined by the language to fully represent the space of values that the enum can take on.  However, [Markus has pointed out before](https://github.com/unicode-org/icu4x/pull/1233#discussion_r740635147) that enumerated Unicode properties can add new definitions for property values over time, that has happened and continues to happen for enumerated properties.  Therefore, we can't represent them using a Rust enum because that would become a breaking change according to how Rust defines enums.

Instead, we have to use the [Newtype](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) idiom.  See `Script` and other similar structs down below for an example.  It behaves almost the same as the enum, but gives us the flexibility to not break our code when we need to represent new enumerated property values in the future.

`CodePointTrie`s can represent data as `u8`, `u16`, or `u32`.  So the newtype is a struct that is a wrapper over an unnamed argument of one of those types. We can find out which type ICU chooses to represent Bidi_Class values by looking at the serialized form of the `CodePointTrieHeader` in the exported data in `bc.toml` that you generated. In particular, we have the field `data_8` for the Bidi_Class CPT data.  The code we have in ICU4X for the properties data provider knows how to check for `data_8` and [then know that the CPT data array values are `u8`](https://github.com/unicode-org/icu4x/blob/main/provider/uprops/src/enum_codepointtrie.rs#L66-L77).

Reply:
suggestion: double check the width of the integer type for `BidiClass`.

The newtype struct `BidiClass` wrapping a `u16`.  Verify & be able to explain why that is the most appropriate type.

Reply:
@echeran , shall I write why I have chosen `u8` in  a comment ?

Reply:
So long as you can follow along and understand everything I wrote here and [the other comment](https://github.com/unicode-org/icu4x/pull/1716#discussion_r835696774), then you don't need to write it out.  Ping me off line if you want to fully verify things with me.